test/test_query2.cpp

```diff
@@ -6056,55 +6056,95 @@ TEST(Query_FullTextMulti)
+        return get_keys(table->query(query).find_all());
+    };
+
+    CHECK_THROW_ANY(do_fulltext_find(""));
 
     // search with multiple terms
-    auto tv = table->where().fulltext(col, "object gemstone").find_all();
-    CHECK_EQUAL(2, tv.size());
+    CHECK_EQUAL(do_fulltext_find("one three"), Keys({7, 8}));
+    CHECK_EQUAL(do_fulltext_find("three one"), Keys({7, 8}));
+    CHECK_EQUAL(do_fulltext_find("C# c++"), Keys({4}));
+    CHECK_EQUAL(do_fulltext_find("1990s"), Keys({2, 4}));
+    CHECK_EQUAL(do_fulltext_find("1990s c++"), Keys({4}));
+    CHECK_EQUAL(do_fulltext_find("object gemstone"), Keys({2, 4}));
 
     // over links
-    tv = origin->link(col_link).column<String>(col).fulltext("object gemstone").find_all();
-    CHECK_EQUAL(2, tv.size());
-
-    tv = origin->query("link.text TEXT 'object gemstone'").find_all();
-    CHECK_EQUAL(2, tv.size());
+    CHECK_EQUAL(do_query_find(origin, "link.text TEXT 'object gemstone'"), Keys({2, 4}));
+    auto tv = origin->link(col_link).column<String>(col).fulltext("object gemstone").find_all();
+    CHECK_EQUAL(get_keys(tv), Keys({2, 4}));
 
     // through LnkLst
     auto obj = tv.get_object(0);
     auto ll = obj.get_linklist(col_link);
     tv = table->where(ll).fulltext(col, "object gemstone").find_all();
-    CHECK_EQUAL(1, tv.size());
+    CHECK_EQUAL(get_keys(tv), Keys({2}));
 
     // Diacritics ignorant
-    tv = table->where().fulltext(col, "depot emanant").find_all();
-    CHECK_EQUAL(1, tv.size());
+    CHECK_EQUAL(do_fulltext_find("depot emanant archive"), Keys({5}));
 
     // search for combination that is not present
-    tv = table->where().fulltext(col, "object data").find_all();
-    CHECK_EQUAL(0, tv.size());
+    CHECK_EQUAL(do_fulltext_find("object data"), Keys());
 
     // exclude words
-    tv = table->where().fulltext(col, "object -databases").find_all();
-    CHECK_EQUAL(4, tv.size());
-    tv = table->where().fulltext(col, "-databases object -duplicates").find_all();
-    CHECK_EQUAL(3, tv.size());
-    tv = table->where().fulltext(col, "object -objects").find_all();
-    CHECK_EQUAL(3, tv.size());
-    CHECK_THROW_ANY(table->where().fulltext(col, "-databases").find_all());
+    CHECK_EQUAL(do_fulltext_find("-three one"), Keys({9}));
+    CHECK_EQUAL(do_fulltext_find("one -three"), Keys({9}));
+    CHECK_EQUAL(do_fulltext_find("object -databases"), Keys({1, 2, 4, 6}));
+    CHECK_EQUAL(do_fulltext_find("-databases object -duplicates"), Keys({1, 2, 4}));
+    CHECK_EQUAL(do_fulltext_find("object -objects"), Keys({2, 4, 6}));
+    CHECK_EQUAL(do_fulltext_find("-object objects"), Keys({3}));
+    CHECK_EQUAL(do_fulltext_find("databases -database"), Keys({}));
+    CHECK_EQUAL(do_fulltext_find("-database databases"), Keys({}));
+    CHECK_EQUAL(do_fulltext_find("database -databases"), Keys({1, 2, 4}));
+    CHECK_EQUAL(do_fulltext_find("-databases database"), Keys({1, 2, 4}));
+
+    // invalid exclude searches
```

Comment:
These are some unexpected results, which i'm not entirely sure are by design or simply bugs. I'd like to merge this to original PR and discuss if something should be fixed or simply documented. Btw, do we have some docs for this feature explaining which version of FTS we intend to support?